crates/ruff_python_formatter/src/comments/visitor.rs

```diff
@@ -683,13 +686,18 @@ impl TraversalSignal {
     }
 }
 
+pub(crate) trait CommentsCollector<'a>: Default {
+    fn add_comment(&mut self, placement: DecoratedComment<'a>, locator: &Locator);
+}
+
 #[derive(Clone, Debug, Default)]
-struct CommentsBuilder<'a> {
+pub(crate) struct CommentsBuilder<'a> {
     comments: CommentsMap<'a>,
 }
 
-impl<'a> CommentsBuilder<'a> {
-    fn add_comment(&mut self, placement: CommentPlacement<'a>) {
+impl<'a> CommentsCollector<'a> for CommentsBuilder<'a> {
+    fn add_comment(&mut self, placement: DecoratedComment<'a>, locator: &Locator) {
+        let placement = place_comment(placement, locator);
```

Comment:
Hmm, this kind of changes the responsibilities I had in mind for the `visitor` and builder. The idea is that the builder is relatively dumb and simply collects the comments.